File: subprojects/dumbbell/src/main/java/com/tambapps/marcel/dumbbell/PulledArtifact.java
package com.tambapps.marcel.dumbbell;

import com.tambapps.maven.dependency.resolver.data.Artifact;
import lombok.Value;

import java.io.File;

@Value
public class PulledArtifact extends Artifact {

  File jarFile;

  public PulledArtifact(Artifact artifact, File jarFile) {
    super(artifact.getGroupId(), artifact.getArtifactId(), artifact.getVersion());
    this.jarFile = jarFile;
  }
}

File: marcel-subprojects/dumbbell-core/src/main/kotlin/com/tambapps/marcel/dumbbell/PulledArtifact.kt
package com.tambapps.marcel.dumbbell

import com.tambapps.maven.dependency.resolver.data.Artifact
import java.io.File
import java.util.*

class PulledArtifact(artifact: Artifact, val jarFile: File?) :
  Artifact(artifact.groupId, artifact.artifactId, artifact.version) {
  override fun toString(): String {
    return "PulledArtifact(artifact=" + toArtifactString() + ", jarFile=" + jarFile + ")"
  }

  override fun equals(o: Any?): Boolean {
    if (this === o) return true
    if (o !is PulledArtifact) return false
    if (!super.equals(o)) return false
    return jarFile == o.jarFile
  }

  override fun hashCode(): Int {
    return Objects.hash(super.hashCode(), jarFile)
  }
}
